util/webhook/webhook.go

```diff
@@ -189,11 +193,31 @@ func affectedRevisionInfo(payloadIf any) (webURLs []string, revision string, cha
 		// See: https://confluence.atlassian.com/bitbucket/event-payloads-740262817.html#EventPayloads-Push
 		// NOTE: this is untested
 		webURLs = append(webURLs, payload.Repository.Links.HTML.Href)
-		// TODO: bitbucket includes multiple changes as part of a single event.
-		// We only pick the first but need to consider how to handle multiple
-		for _, change := range payload.Push.Changes {
-			revision = change.New.Name
-			break
+		for _, changes := range payload.Push.Changes {
+			revision = changes.New.Name
+			change.shaBefore = changes.Old.Target.Hash
+			change.shaAfter = changes.New.Target.Hash
+			var repoCreds *v1alpha1.RepoCreds
+			var err error
+			// when webhook.bitbucket.secret is set in argocd-secret, then the payload must be signed and
+			// signature is validated before payload is parsed. Get DiffSet only for authenticated webhooks.
+			if len(a.settings.WebhookBitbucketServerSecret) > 0 {
```

Comment:
I think this should be BitbucketUUID instead of BitbucketServerSecret. But the idea lgmt.

Reply:
Corrected. Made the required changes in the recent commit.